test_provider.py

```diff
@@ -0,0 +1,310 @@
+                if json[key]['base']['tag'] == 'sidereal':
+                    target = JsonProvider.parse_sidereal_target(json[key])
+                else:
+                    target = JsonProvider.parse_sidereal_target(json[key])
+                targets.append(target)
+
+                for guide_group in json[key]['guideGroups']:
+                    
+                    if type(guide_group[1]) == dict:
+                        guide_stars[target.name] = JsonProvider.parse_guide_star(guide_group[1])
+        
+        find_constraints = [json[key] for key in json.keys() if key.startswith('SCHEDULING_CONDITIONS')]
+        constraints = JsonProvider.parse_constraints(find_constraints[0]) if len(find_constraints) > 0 else None
+                
+        qa_states = [QAState(JsonProvider.qa_states[log_entry['qaState']]) for log_entry in json['obsLog']]
+
+        site = Site.GN if json['observationId'].split('-')[0] == 'GN' else Site.GS
+        status = ObservationStatus(json['obsStatus'])
+        priority = Priority.HIGH if json['priority'] == 'HIGH' else (Priority.LOW if json['priority'] == 'LOW' else Priority.MEDIUM)
+        print('observationId: ', json['observationId'])
+        atoms = JsonProvider.parse_atoms(json['sequence'], qa_states)
+
+        obs = Observation(json['observationId'],
+                          json['key'],
+                          int(name.split('-')[1]),
+                          json['title'],
+                          site,
+                          status,
+                          None,
+                          priority,
+                          SetupTimeType(json['setupTimeType']),
+                          timedelta(milliseconds=json['setupTime']),
+                          JsonProvider.obs_classes[json['obsClass']],
+                          None,
+                          None,
+                          None,
+                          targets,
+                          guide_stars,
+                          atoms,
+                          constraints,
+                          None
+                          )
+
+
+        # obs.active = ...
+
+        # obs.exec_time = ... # Total time sequence  + overhead
+        # obs.program_used = ...
+        # obs.partner_used = ...
+       
+        # obs.too_type = ... # TODO: at program level
+
+        return obs
+    
+    @staticmethod
+    def parse_time_allocation(json: dict) -> TimeAllocation:
+        #print(json['timeAccountAllocationCategories'])
+        ta = TimeAllocation(TimeAccountingCode(json['timeAccountAllocationCategories'][0]['category']),
+                            timedelta(milliseconds=json['awardedTime']),
+                            timedelta(milliseconds=0),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['programTime']),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['partnerTime']))
+        return ta
```

Comment:
More places where we should remove hardcoded `str`s.

Also, why are we just looking at index `[0]`? I'm not clear exactly what is happening here.